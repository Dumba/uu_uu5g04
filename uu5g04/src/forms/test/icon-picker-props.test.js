/**
 * Copyright (C) 2021 Unicorn a.s.
 *
 * This program is free software: you can redistribute it and/or modify it under the terms of the GNU General Public
 * License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later
 * version.
 *
 * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied
 * warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License at
 * <https://gnu.org/licenses/> for more details.
 *
 * You may obtain additional information at <https://unicorn.com> or contact Unicorn a.s. at address: V Kapslovne 2767/2,
 * Praha 3, Czech Republic or at the email: info@unicorn.com.
 */

//@@viewOn:imports
import UU5 from "uu5g04";
import "uu5g04-bricks";
import "uu5g04-forms";
//@@viewOff:imports

const { mount, shallow, wait } = UU5.Test.Tools;

const MixinPropsFunction = UU5.Common.VisualComponent.create({
  mixins: [UU5.Common.BaseMixin],

  getInitialState: () => {
    return {
      isCalled: false,
      value: "",
      message: "",
      feedback: "initial",
    };
  },
  onChangeHandler(event) {
    alert("onChange event has been called.");
    this.setState({ isCalled: true });
    this.setState({ value: event.target.value });
    this.setState({ message: "Is valid." });
    this.setState({ feedback: "success" });
  },

  onValidateHandler(event) {
    alert("onValidate event has been called.");
    this.setState({ isCalled: true });
    this.setState({ value: event.target.value });
    this.setState({ message: "Is valid." });
    this.setState({ feedback: "success" });
  },

  onChangeFeedbackHandler(event) {
    alert("onChangeFeedback event has been called.");
    this.setState({ isCalled: true });
    this.setState({ value: event.target.value });
    this.setState({ message: "Is valid." });
    this.setState({ feedback: "success" });
  },

  render() {
    return (
      <UU5.Forms.IconPicker
        id={"checkID"}
        value={this.state.value}
        feedback={this.state.feedback}
        message={this.state.message}
        onChange={this.onChangeHandler}
        onValidate={this.onValidateHandler}
        onChangeFeedback={this.onChangeFeedbackHandler}
      />
    );
  },
});

//`UU5.Forms.IconPicker`
const CONFIG = {
  mixins: [
    "UU5.Common.BaseMixin",
    "UU5.Common.ElementaryMixin",
    "UU5.Common.PureRenderMixin",
    "UU5.Common.ScreenSizeMixin",
    "UU5.Forms.InputMixin",
    "UU5.Common.ColorSchemaMixin",
  ],
  props: {
    categories: {
      values: [["mdi", "fa", "glyphicon", "uu5"], ["plus4u", "plus4u5"], undefined],
    },
    selectedCategory: {
      values: ["mdi", "fa", "glyphicon", "uu5", "plus4u", "plus4u5", "all"],
      requiredProps: {
        categories: ["mdi", "fa", "glyphicon", "uu5", "plus4u", "plus4u5"],
      },
    },
    placeholder: {
      value: ["placeholder"],
    },
    disableBackdrop: {
      values: [true, false],
    },
  },
  opt: {
    shallowOpt: {
      disableLifecycleMethods: false,
    },
  },
};

describe(`UU5.Forms.IconPicker props`, () => {
  UU5.Test.Tools.testProperties(UU5.Forms.IconPicker, CONFIG);
});

describe(`UU5.Forms.IconPicker props function -> InputMixin`, () => {
  it("onChange()", () => {
    window.alert = jest.fn();
    const wrapper = shallow(<MixinPropsFunction />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.instance().state.isCalled).toBeFalsy();
    expect(wrapper.instance().state.value).toEqual("");
    expect(wrapper.instance().state.message).toEqual("");
    expect(wrapper.instance().state.feedback).toMatch(/initial/);
    wrapper.simulate("change", { target: { value: "Testing react in jest" } });
    expect(window.alert).toBeCalled();
    expect(window.alert).toHaveBeenCalledWith("onChange event has been called.");
    expect(window.alert.mock.calls[0][0]).toEqual("onChange event has been called.");
    expect(wrapper.instance().state.isCalled).toBeTruthy();
    expect(wrapper.instance().state.value).toEqual("Testing react in jest");
    expect(wrapper.instance().state.message).toEqual("Is valid.");
    expect(wrapper.instance().state.feedback).toEqual("success");
    expect(wrapper).toMatchSnapshot();
  });

  it(`onChangeDefault() with callback`, () => {
    let callback = jest.fn();
    let wrapper = shallow(<UU5.Forms.IconPicker />);
    wrapper.instance().onChangeDefault({}, callback);
    expect(callback).toBeCalled();
  });

  it("onValidate()", () => {
    window.alert = jest.fn();
    const wrapper = shallow(<MixinPropsFunction />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.instance().state.isCalled).toBeFalsy();
    expect(wrapper.instance().state.value).toEqual("");
    expect(wrapper.instance().state.message).toEqual("");
    expect(wrapper.instance().state.feedback).toMatch(/initial/);
    wrapper.simulate("validate", { target: { value: "Testing react in jest" } });
    expect(window.alert).toBeCalled();
    expect(window.alert).toHaveBeenCalledWith("onValidate event has been called.");
    expect(window.alert.mock.calls[0][0]).toEqual("onValidate event has been called.");
    expect(wrapper.instance().state.isCalled).toBeTruthy();
    expect(wrapper.instance().state.value).toEqual("Testing react in jest");
    expect(wrapper.instance().state.message).toEqual("Is valid.");
    expect(wrapper.instance().state.feedback).toEqual("success");
    expect(wrapper).toMatchSnapshot();
  });

  it("onChangeFeedback()", () => {
    window.alert = jest.fn();
    const wrapper = shallow(<MixinPropsFunction />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.instance().state.isCalled).toBeFalsy();
    expect(wrapper.instance().state.value).toEqual("");
    expect(wrapper.instance().state.message).toEqual("");
    expect(wrapper.instance().state.feedback).toMatch(/initial/);
    wrapper.simulate("changeFeedback", { target: { value: "Testing react in jest" } });
    expect(window.alert).toBeCalled();
    expect(window.alert).toHaveBeenCalledWith("onChangeFeedback event has been called.");
    expect(window.alert.mock.calls[0][0]).toEqual("onChangeFeedback event has been called.");
    expect(wrapper.instance().state.isCalled).toBeTruthy();
    expect(wrapper.instance().state.value).toEqual("Testing react in jest");
    expect(wrapper.instance().state.message).toEqual("Is valid.");
    expect(wrapper.instance().state.feedback).toEqual("success");
    expect(wrapper).toMatchSnapshot();
  });
});

describe(`UU5.Forms.IconPicker default props`, () => {
  it(`UU5.Forms.IconPicker check default props`, () => {
    const wrapper = shallow(<UU5.Forms.IconPicker id={"uuID"} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.instance().props.value).toBe(null);
    expect(wrapper.instance().props.categories).toEqual(["mdi"]);
    expect(wrapper.instance().props.selectedCategory).toBe("mdi");
    expect(wrapper.instance().props.disableBackdrop).toBeFalsy();
    expect(wrapper.instance().props.placeholder).toBe(null);
  });
});

describe(`UU5.Forms.IconPicker check default default props from Mixins`, () => {
  it(`UU5.Forms.InputMixin`, () => {
    const wrapper = shallow(<UU5.Forms.IconPicker id={"uuID"} />);
    expect(wrapper.instance().props.inputAttrs).toBe(null);
    expect(wrapper.instance().props.size).toEqual("m");
    expect(wrapper.instance().props.readOnly).toBeFalsy();
    expect(wrapper.instance().props.feedback).toEqual("initial");
    expect(wrapper.instance().props.message).toBe(null);
    expect(wrapper.instance().props.label).toBe(null);
    expect(wrapper.instance().props.onChange).toBe(null);
    expect(wrapper.instance().props.onValidate).toBe(null);
    expect(wrapper.instance().props.onChangeFeedback).toBe(undefined);
    expect(wrapper.instance().props.inputColWidth).toMatchObject({ xs: 12, s: 7 });
    expect(wrapper.instance().props.labelColWidth).toMatchObject({ xs: 12, s: 5 });
  });

  it(`UU5.Common.Base,Elementary,Pure,Color`, () => {
    const wrapper = shallow(<UU5.Forms.IconPicker id={"uuID"} />);
    //Check UU5.Common.Elementary.Mixin default props
    expect(wrapper.instance().props.hidden).toBeFalsy();
    expect(wrapper.instance().props.disabled).toBeFalsy();
    expect(wrapper.instance().props.selected).toBeFalsy();
    expect(wrapper.instance().props.controlled).toBeTruthy;
    //Check UU5.Common.PureRender.Mixin default values
    expect(wrapper.instance().props.pureRender).toBeFalsy();
    //Check default values of props BaseMixin.
    expect(wrapper.instance().props.id).toEqual("uuID");
    expect(wrapper.instance().props.name).toBe(null);
    expect(wrapper.instance().props.tooltip).toBe(null);
    expect(wrapper.instance().props.className).toBe(null);
    expect(wrapper.instance().props.style).toBe(null);
    expect(wrapper.instance().props.mainAttrs).toBe(null);
    expect(wrapper.instance().props.parent).toBe(null);
    expect(wrapper.instance().props.ref_).toBe(null);
    expect(wrapper.instance().props.noIndex).toBeFalsy();
    //default value of colorSchema and LSI language props are in snapshot.
  });
});
