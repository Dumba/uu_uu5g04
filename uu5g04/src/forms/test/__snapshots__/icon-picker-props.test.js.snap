// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UU5.Forms.IconPicker default props UU5.Forms.IconPicker check default props 1`] = `
<div
  className="uu5-forms-iconpicker uu5-forms-input uu5-forms-input-m uu5-forms-input-initial uu5-forms-input-prevent-wrap"
  id="uuID"
  style={Object {}}
  title={null}
>
  <InputWrapper
    colWidth="uu5-col-xs12 uu5-col-s12 uu5-col-m12 uu5-col-l12 uu5-col-xl12"
    disabled={false}
    readonly={false}
  >
    <Div
      className="uu5-forms-right-wrapper"
      key="button"
    >
      <Button
        className="uu5-forms-iconpicker-open-button"
        id="uuID-button"
        onClick={[Function]}
        ref_={[Function]}
      >
         
        <Icon
          className="uu5-forms-iconpicker-arrow-down-icon"
          icon="mdi-menu-down"
        />
      </Button>
    </Div>
    <forwardRef(Popover)
      className=""
      controlled={false}
      disableBackdrop={false}
      fitHeightToViewport={true}
      forceRender={true}
      header={
        <Row
          className={null}
          content={null}
          controlled={true}
          disabled={false}
          display="standard"
          dynamic={false}
          footer={null}
          header={null}
          hidden={false}
          id={null}
          ignoreInnerHTML={false}
          increaseLevel={false}
          level={null}
          mainAttrs={null}
          name={null}
          nestingLevel={null}
          noIndex={false}
          noSpacing={false}
          parent={null}
          pureRender={false}
          ref_={null}
          selected={false}
          style={null}
          tooltip={null}
          underline={false}
        >
          <TextIcon
            actionOnEnter={false}
            autocompleteItems={null}
            bgStyle={null}
            borderRadius={null}
            className="uu5-forms-iconpicker-search-input"
            colorSchema={null}
            controlled={true}
            disabled={false}
            elevation={null}
            feedback="initial"
            focusMessage={null}
            hidden={false}
            icon="mdi-magnify"
            iconPosition="right"
            id={null}
            inputAttrs={null}
            inputColWidth={
              Object {
                "s": 7,
                "xs": 12,
              }
            }
            inputWidth={null}
            label={null}
            labelColWidth={
              Object {
                "s": 5,
                "xs": 12,
              }
            }
            labelWidth={null}
            mainAttrs={null}
            message={null}
            name={null}
            noIndex={false}
            onBlur={null}
            onChange={[Function]}
            onClick={null}
            onEnter={null}
            onFocus={null}
            onValidate={null}
            parent={null}
            password={false}
            pattern={null}
            patternMessage={null}
            placeholder="Search icon"
            popoverLocation="local"
            pureRender={false}
            readOnly={false}
            ref_={null}
            required={false}
            requiredMessage={null}
            selected={false}
            size="s"
            spacing={null}
            style={null}
            tooltip={null}
            tooltipIcon="mdi-information-outline"
            validateOnChange={false}
            value=""
          />
        </Row>
      }
      id="uuID-popover"
      key="popover"
      location="local"
      ref_={[Function]}
    />
  </InputWrapper>
</div>
`;

exports[`UU5.Forms.IconPicker props function -> InputMixin onChange() 1`] = `
<IconPicker
  id="checkID"
  message=""
  onChange={[Function]}
  onChangeFeedback={[Function]}
  onValidate={[Function]}
  value=""
/>
`;

exports[`UU5.Forms.IconPicker props function -> InputMixin onChange() 2`] = `
<IconPicker
  feedback="success"
  id="checkID"
  message="Is valid."
  onChange={[Function]}
  onChangeFeedback={[Function]}
  onValidate={[Function]}
  value="Testing react in jest"
/>
`;

exports[`UU5.Forms.IconPicker props function -> InputMixin onChangeFeedback() 1`] = `
<IconPicker
  id="checkID"
  message=""
  onChange={[Function]}
  onChangeFeedback={[Function]}
  onValidate={[Function]}
  value=""
/>
`;

exports[`UU5.Forms.IconPicker props function -> InputMixin onChangeFeedback() 2`] = `
<IconPicker
  feedback="success"
  id="checkID"
  message="Is valid."
  onChange={[Function]}
  onChangeFeedback={[Function]}
  onValidate={[Function]}
  value="Testing react in jest"
/>
`;

exports[`UU5.Forms.IconPicker props function -> InputMixin onValidate() 1`] = `
<IconPicker
  id="checkID"
  message=""
  onChange={[Function]}
  onChangeFeedback={[Function]}
  onValidate={[Function]}
  value=""
/>
`;

exports[`UU5.Forms.IconPicker props function -> InputMixin onValidate() 2`] = `
<IconPicker
  feedback="success"
  id="checkID"
  message="Is valid."
  onChange={[Function]}
  onChangeFeedback={[Function]}
  onValidate={[Function]}
  value="Testing react in jest"
/>
`;
